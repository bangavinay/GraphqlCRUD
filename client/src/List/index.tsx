import React from 'react';
import { useHistory } from "react-router-dom";
import { GET_POSTS_QUERY, DELETE_POSTS_MUTATION } from '../Queries';


const List = () => {
    const { loading, error, data,refetch } = GET_POSTS_QUERY();
    const [deletePost] = DELETE_POSTS_MUTATION();
    const history = useHistory();
    // refetch();

    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error :(</p>;
    let editPost = (id) =>{
      history.push(`/edit/${id}`);
    }
    return data.posts.map(({ id, title, description, active }) => {
      if(active){
        return (<div key={id}>
        <div className="post">
          <h3>{title}</h3> 
          <p>{description}</p>
          <button onClick={()=>editPost(id)}> Edit</button>
          <button onClick={e => { deletePost({ variables: { id } });refetch();}} >Delete</button>
        </div>
      </div>);
      }
    });
  };

  export default List;