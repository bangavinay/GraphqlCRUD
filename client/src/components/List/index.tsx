import React from 'react';
import { useHistory } from "react-router-dom";
import { GET_POSTS_QUERY, DELETE_POSTS_MUTATION } from '../../queries';


const List = () => {
    const [ deletePost ] = DELETE_POSTS_MUTATION();
    const history = useHistory();

    //Fetching the list of posts 
    const { loading, error, data, refetch } = GET_POSTS_QUERY();
    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error :(</p>;

    //Routing to Edit page on click of edit button  
    const  editPost = (id: number) => {
      history.push(`/edit/${id}`);
    }
    return data.posts.map(({ id, title, description, active }) => {
      if(active){
        return (<div key={id}>
        <div className="post">
          <h3>{title}</h3> 
          <p>{description}</p>
          <button onClick={()=>editPost(id)}> Edit</button>
          <button onClick={() => { deletePost({ variables: { id } });refetch();}} >Delete</button>
        </div>
      </div>);
      }
    });
  };

  export default List;