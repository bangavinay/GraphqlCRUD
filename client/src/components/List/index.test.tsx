import React from 'react';
import { render, wait } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import List from '.';

import { GET_POSTS } from '../../queries';


describe('Render Edit', () => {
    const postMock1 = [{
        request: {
          query: GET_POSTS,
          variables: { },
        },
        result: {
            data: {
              posts: [
                { 
                  id: 1, 
                  title: "Buck", 
                  description: "bulldog", 
                  active:true 
                },
                { 
                  id: 2,
                  title: "hh",
                  description: "bulldog",
                  active: true 
                }
              ],
            },
            refetch: jest.fn(),
          },
      }];
      
    test('Loading State of List ', async () => {
        const  component =render(
            <MockedProvider mocks={postMock1} addTypename={false}>
                <List  />
            </MockedProvider>);

        await wait();
        expect(component).toMatchSnapshot()
    });
});
