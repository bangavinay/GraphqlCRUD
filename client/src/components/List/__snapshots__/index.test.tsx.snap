// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Edit Loading State of List  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="post"
        >
          <h3>
            Buck
          </h3>
          <p>
            bulldog
          </p>
          <button>
             Edit
          </button>
          <button>
            Delete
          </button>
        </div>
      </div>
      <div>
        <div
          class="post"
        >
          <h3>
            hh
          </h3>
          <p>
            bulldog
          </p>
          <button>
             Edit
          </button>
          <button>
            Delete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="post"
      >
        <h3>
          Buck
        </h3>
        <p>
          bulldog
        </p>
        <button>
           Edit
        </button>
        <button>
          Delete
        </button>
      </div>
    </div>
    <div>
      <div
        class="post"
      >
        <h3>
          hh
        </h3>
        <p>
          bulldog
        </p>
        <button>
           Edit
        </button>
        <button>
          Delete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
